File: Data Structures & Algorithms/anagram-groups/submission-8.py
from collections import defaultdict
class Solution:
    def groupAnagrams(self, strs: List[str]) -> List[List[str]]:
        d = defaultdict(list)
        for s in strs:
            freq = [0] * 26
            for c in s:
                freq[ord(c) - ord('a')] += 1
            d[tuple(freq)].append(s)
        return list(d.values())

File: Data Structures & Algorithms/is-anagram/submission-8.cpp
class Solution {
public:
    bool isAnagram(string s, string t) {
        if (s.size() != t.size()) return false;
        vector<int> f(26);
        for (char c : s){
            f[c - 'a']++;
        }
        for (char c : t){
            f[c - 'a']--;
            if (f[c - 'a'] < 0) return false;
        }
        return true;
    }
};
